notebooks/02_model/lightfm_deep_dive.ipynb

```diff
@@ -0,0 +1,1505 @@
+{
```

Comment:
<p>copying same comment here:</p><p>for consistency it would be good to use our own precision_at_k and other metrics. One example that would be very similar to your is&nbsp;<a href="https://github.com/microsoft/recommenders/blob/master/notebooks/02_model/surprise_svd_deep_dive.ipynb" target="_blank">https://github.com/microsoft/recommenders/blob/master/notebooks/02_model/surprise_svd_deep_dive.ipynb</a>. There we are using a similar approach, using an external library, called surprise, and try to add as less code as needed to show how it works</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1096/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Reply:
Hi @miguelgfierro , thanks for your comments. I agree with you on this point using the existing methods to extract metrics. 
The difficulty being LightFM doesn't predict the user-item ratings directly (the prediction scores are only for ranking purposes), hence it would involve rather significant modification to get it to work with the existing codes.
Let me know how do you want to progress this/if you still think this is critical?
Thanks

Reply:
there has to be a mapping between their ranking metrics and ours. 

It is in fact important, the reason is because that enable us to compare between algos. But obviously, it depends whether you have bandwidth or not. In case, you don't have bandwidth, I would prefer to not show the metrics, so we don't cause confusion

Reply:
Hi there, a bit puzzled in how to get built-in evaluation metrics to work - if I create a sample test df with all predictions = 1 (for argument sage), I get a 0.886 Prec@K - see following: 

![Screenshot from 2020-05-05 19-26-24](https://user-images.githubusercontent.com/47578869/81101758-814b6880-8f06-11ea-8027-52c1094cba62.png)

However, if I assign the prediction column to random ints, I also get exactly the same Prec@K - see attached:

![Screenshot from 2020-05-05 19-30-53](https://user-images.githubusercontent.com/47578869/81102077-f9b22980-8f06-11ea-9b1d-9695113da7b5.png)

In fact have similar issues with NDCG and Recall@K as well, so not exactly sure whether I am using these methods correctly. Would be grateful if you could advise.

Thanks

Reply:
@yueguoguo worked on the metrics, he might be able to help

Reply:
Thanks @miguelgfierro, I've solved that and I'll try to re-structure the deep dive notebook differently later.

Reply:
awesome! I'll close this item then